langchain4j/src/main/java/dev/langchain4j/data/document/source/ClassPathSource.java

```diff
@@ -0,0 +1,34 @@
+package dev.langchain4j.data.document.source;
+
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+
+/**
+ * Specialization of a {@link FileSystemSource} that knows how to read from the classpath
+ */
+public class ClassPathSource extends FileSystemSource {
+    public ClassPathSource(Path path) {
+        super(path);
+    }
+
+    public static ClassPathSource fromClasspath(String classPathResource) {
```

Comment:
By Making the return type as Optional is more controllabe

Reply:
I'm not sure what you mean? I followed the same pattern that `FileSystemSource` does - throw and exception if the URL is null.

Reply:
my fault
looked at the PR with the phone 🤪